### PR TITLE
refactor: fix chromium-style warnings in linux code

### PR DESCRIPTION
**Pull request by nornagon**

### CODE REVIEW

**Comment by nornagon** [comment]
@ckerr the new results look spurious (something to do with vendor/pdf_viewer?)

File: atom/browser/native_browser_view_views.h [hunk 1]
```diff
@@ -16,9 +16,7 @@ class NativeBrowserViewViews : public NativeBrowserView {
   ~NativeBrowserViewViews() override;
 
   uint8_t GetAutoResizeFlags() { return auto_resize_flags_; }
-  void SetAutoResizeFlags(uint8_t flags) override {
-    auto_resize_flags_ = flags;
-  }
+  void SetAutoResizeFlags(uint8_t flags) override;
```
Comment (ckerr) [inline atom/browser/native_browser_view_views.h]: not a blocker, but what was the issue with not inlining this? size-would-be-larger warning?

File: atom/browser/osr/osr_render_widget_host_view.h [hunk 2]
```diff
@@ -184,6 +184,7 @@ class OffScreenRenderWidgetHostView
   std::unique_ptr<ui::CompositorLock> GetCompositorLock(
       ui::CompositorLockClient* client) override;
   void CompositorResizeLockEnded() override;
+  bool IsAutoResizeEnabled() const override;
```
Comment (ckerr) [inline atom/browser/osr/osr_render_widget_host_view.h]: Was this not needed on non-macOS platforms?

File: atom/browser/ui/views/menu_bar.cc [hunk 3]
```diff
@@ -26,7 +26,7 @@ const SkColor kDefaultColor = SkColorSetARGB(255, 233, 233, 233);
 }  // namespace
 
 MenuBar::MenuBar(views::View* window)
-    : background_color_(kDefaultColor), menu_model_(NULL), window_(window) {
+    : background_color_(kDefaultColor), window_(window), menu_model_(NULL) {
```
Comment (ckerr) [inline atom/browser/ui/views/menu_bar.cc]: genuinely surprised that some of these, like order-of-initialization, weren't already warnings

File: atom/browser/native_browser_view_views.h (hunk 1 above)
Comment (nornagon) [reply atom/browser/native_browser_view_views.h]: Virtual methods are never actually inlined: https://www.chromium.org/developers/coding-style/chromium-style-checker-errors#TOC-Virtual-Method-Out-of-lining

File: atom/browser/osr/osr_render_widget_host_view.h (hunk 2 above)
Comment (nornagon) [reply atom/browser/osr/osr_render_widget_host_view.h]: It's _only_ needed on non-macOS platforms, as far as I can tell. On macOS, it doesn't override anything in the parent class and we never call it.

**Comment by nornagon** [comment]
I don't think removing `virtual` is an optimization at all; `override` implies `virtual` and the chromium-style checker just requires that you don't specify both.

**Pull request merged**